programs/eosio-launcher/main.cpp

```diff
@@ -605,8 +615,17 @@ launcher_def::initialize (const variables_map &vmap) {
     }
   }
 
-  config_dir_base = "etc/eosio";
-  data_dir_base = "var/lib";
+  config_dir_base_cluster0 = "etc/eosio";
+  if (cluster_id == 0) {
+    config_dir_base = config_dir_base_cluster0;
+    data_dir_base = "var/lib";
+  } else {
+    std::stringstream config_s, data_s;
```

Comment:
It is faster to just convert the number to a string and then append it to the existing string than use `std::stringstream` here.

Also when converting the number to a string it would be good to always have 5 digits with padded zeros as necessary. That way lexicographical sorting of the directories would always work correctly.